upgrade/scripts/upgrade/prerequisites.sh

```diff
@@ -756,6 +756,22 @@ if [[ ${state_recorded} == "0" && $(hostname) == "${PRIMARY_NODE}" ]]; then
       fi
     fi
 
+    # cert-manager will need to be upgraded if cray-drydock version is less than 2.18.4.
+    # This will only be the case in some CSM 1.6 to CSM 1.6 upgrades.
+    # It only needs to be checked if cert-manager is not already being upgraded.
+    if [ "${needs_upgrade}" -eq 0 ]; then
+      drydock_vers=$(helm get metadata -n loftsman cray-drydock -o json | jq -r '.version')
+      major=2
+      minor=18
+      patch=4
+      if [[ $(echo "$drydock_vers" | cut -d. -f1) -eq $major ]] && [[ $(echo "$drydock_vers" | cut -d. -f2) -eq $minor ]] && [[ $(echo "$drydock_vers" | cut -d. -f3) -lt $patch ]]; then
```

Comment:
nit: this is hard to read. Pulling the `$(echo "$drydock_vers" | cut -d. -f)` commands out would make the conditional easier.

```suggestion
      drydock_major="${drydock_vers%%.*}"
      drydock_minor_patch="${drydock_vers#*.}" # temp
      drydock_patch="${drydock_minor_patch#*.}"
      drydock_minor="${drydock_minor_patch%.*}"
      if [ $drydock_major -eq $major ] && [ $drydock_minor -eq $minor ] && [ $drydock_patch -lt $patch ]; then
```

The conditional is a little tight, I figure we won't update cray-drydock's major or minor version but if we did then this would break.

Consider using something like this to prevent needing to fix this as we branch release/1.7 and onwards:

```bash
if [ $drydock_major -lt $major ]; then
  needs_upgrade=1
elif [ $drydock_major -eq $major ] && [ $drydock_minor -lt $minor ]; then
  needs_upgrade=1
elif [ $drydock_major -eq $major ] && [ $drydock_minor -eq $minor ] && [ $drydock_patch -lt $patch ]; then
  needs_upgrade=1
fi
if [ $needs_upgrade -ne 0 ]; then
  echo "'cray-drydock version [$drydock_vers] less than $major.$minor.$patch and needs to be upgraded."
fi
```

Reply:
I have made these changes, thanks